vscode-dotnet-runtime-library/src/Acquisition/ExistingPathResolver.ts

```diff
@@ -22,15 +23,22 @@ If you would like to continue to use the setting anyways, set dotnetAcquisitionE
 
 export class ExistingPathResolver
 {
+    private lastSeenNonTruePathValue: string | null = null;
 
-    public constructor(private readonly workerContext : IAcquisitionWorkerContext, private readonly utilityContext : IUtilityContext, private executor? : ICommandExecutor)
+    public constructor(private readonly workerContext: IAcquisitionWorkerContext, private readonly utilityContext: IUtilityContext, private executor?: ICommandExecutor)
     {
         this.executor ??= new CommandExecutor(this.workerContext, this.utilityContext);
     }
 
-    public async resolveExistingPath(existingPaths: IExistingPaths | undefined, extensionId: string | undefined, windowDisplayWorker: IWindowDisplayWorker, requirement? : DotnetVersionSpecRequirement): Promise<IDotnetAcquireResult | undefined>
+    public async resolveExistingPath(existingPaths: IExistingPaths | undefined, extensionId: string | undefined, windowDisplayWorker: IWindowDisplayWorker, requirement?: DotnetVersionSpecRequirement): Promise<IDotnetAcquireResult | undefined>
     {
-        const existingPath = this.getExistingPath(existingPaths, extensionId, windowDisplayWorker);
+        let existingPath = this.getExistingPath(existingPaths, extensionId, windowDisplayWorker);
+        this.lastSeenNonTruePathValue = existingPath;
+        if (!this.allowInvalidPath(this.workerContext))
+        {
+            // The user path setting may be a symlink but we dont want to resolve it since a snap symlink isnt a real directory, we can find the true path better this way:
+            existingPath = (await new DotnetPathFinder(this.workerContext, this.utilityContext, this.executor).getTruePath([existingPath ?? '']))?.at(0) ?? null;
```

Comment:
It seems like getTruePath would return [existingPath] if it doesn't find any real paths. If we don't want invalid paths, is this ok?

Reply:
It will not return anything if it's an invalid path. We also validate the path later, so this should be ok. Thank you for reviewing my code :)